### PR TITLE
Login rib

### PR DESCRIPTION
**Pull request by antonshilov**
**Description**:
Add login rib that launches CustomChromeTab

**Related issue(s)**:

### CODE REVIEW

File: app-example/src/main/java/com/badoo/ribs/example/root/RootInteractor.kt [hunk 1]
```diff
@@ -28,10 +29,8 @@ internal class RootInteractor(
 
     override fun onCreate(nodeLifecycle: Lifecycle) {
         nodeLifecycle.createDestroy {
-            bind(
-                authDataSource
-                    .authUpdates.observeOn(AndroidSchedulers.mainThread()) to authStatConsumer
-            )
+            bind(authDataSource.authUpdates.observeOn(AndroidSchedulers.mainThread()) to authStateConsumer)
```
Comment (zsoltk) [inline app-example/src/main/java/com/badoo/ribs/example/root/RootInteractor.kt:32]:
```suggestion
            bind(authUpdates to authStateConsumer)
```

Maybe extract?

File: app-example/src/main/java/com/badoo/ribs/example/network/ApiFactory.kt [hunk 2]
```diff
@@ -11,16 +14,22 @@ import retrofit2.converter.moshi.MoshiConverterFactory
...
+            client = okhttpClient(isDebug, accessKey, authStateStorage, networkErrorConsumer),
             baseUrl = BASE_URL
         ).create(UnsplashApi::class.java)
 
     private fun okhttpClient(
         isDebug: Boolean,
         accessKey: String,
-        authStateStorage: AuthStateStorage
+        authStateStorage: AuthStateStorage,
+        networkErrorConsumer: Consumer<NetworkError>
```
Comment (zsoltk) [inline app-example/src/main/java/com/badoo/ribs/example/network/ApiFactory.kt:32]: Is this used for anything?

File: app-example/src/main/java/com/badoo/ribs/example/login/feature/LoginFeature.kt [hunk 3]
```diff
@@ -1,29 +1,22 @@
...
 import com.badoo.ribs.example.login.AuthCodeDataSource
-import com.badoo.ribs.example.login.feature.LoginFeature.Effect
-import com.badoo.ribs.example.login.feature.LoginFeature.News
 import com.badoo.ribs.example.login.feature.LoginFeature.State
 import com.badoo.ribs.example.login.feature.LoginFeature.Wish
 import io.reactivex.Observable
-import io.reactivex.Observable.empty
 import io.reactivex.android.schedulers.AndroidSchedulers
 
 internal class LoginFeature(
```
Comment (zsoltk) [inline app-example/src/main/java/com/badoo/ribs/example/login/feature/LoginFeature.kt:13]: This class looks pretty empty

File: app-example/src/main/java/com/badoo/ribs/example/network/ApiFactory.kt (hunk 2 above)
Comment (antonshilov) [reply app-example/src/main/java/com/badoo/ribs/example/network/ApiFactory.kt:32]: Yes `UnauthorizedErrorInterceptor` publishes events to the consumer

File: app-example/src/main/java/com/badoo/ribs/example/login/feature/LoginFeature.kt [hunk 4]
```diff
@@ -1,49 +1,73 @@
...
     initialState = State(),
-    bootstrapper = BootStrapperImpl(authCodeDataSource, authDataSource),
+    bootstrapper = BootStrapperImpl(authCodeDataSource),
+    actor = ActorImpl(authDataSource),
     reducer = ReducerImpl()
 ) {
 
     data class State(
-        val yourData: Any? = null
+        val isLoading: Boolean = false,
```
Comment (zsoltk) [inline app-example/src/main/java/com/badoo/ribs/example/login/feature/LoginFeature.kt:25]:
```suggestion
        val inProgress: Boolean = false,
```

File: app-example/src/main/java/com/badoo/ribs/example/login/feature/LoginFeature.kt [hunk 5]
```diff
@@ -1,49 +1,73 @@
...
 
     sealed class Wish {
-        object AuthSuccess : Wish()
+        class Login(val authCode: String) : Wish()
+    }
+
+    sealed class Effect {
+        object AuthSuccess : Effect()
+        object LoadingStarted : Effect()
+        class LoadingFailed(val error: Throwable?) : Effect()
```
Comment (zsoltk) [inline app-example/src/main/java/com/badoo/ribs/example/login/feature/LoginFeature.kt:36]:
```suggestion
        object AuthInProgress : Effect()
        class AuthFailed(val error: Throwable?) : Effect()
```

**Pull request merged**